library.go

```diff
@@ -1827,6 +1826,49 @@ func string_split(fnname string, recv_ Value, args Tuple, kwargs []Tuple) (Value
 	return NewList(list), nil
 }
 
+func reverse(s string, size int) string {
+	rev := make([]rune, 0, size)
+
+	for _, point := range s {
+		rev = append(rev, point)
+	}
+
+	for i := 0; i < size/2; i++ {
+		rev[i], rev[size-1-i] = rev[size-1-i], rev[i]
+	}
+
+	return string(rev)
```

Comment:
This allocates a copy as large as the input.  How hard would it be to avoid materializing the reversed list?

It should be possible to discover the index of each new code point by working back at most 4 bytes from the previous index.

Reply:
Since I will just move this into the function, I can iterate through the rune slice which should be safe to do. Please correct me if I am wrong.